shoes-core/lib/shoes/slot.rb

```diff
@@ -183,15 +183,19 @@ def position_contents
 
     def positioning(element, current_position)
       if element.needs_positioning?
-        position_element element, current_position
-        element.contents_alignment(current_position) if element.respond_to? :contents_alignment
-      end
+        if element.attached_to
+          align_position = position_attached_element element
+        else
+          position_element element, current_position
+          align_position = current_position
+        end
```

Comment:
Yeah, the lack of symmetry between attached/not there was bothering me. Will see what I can work up.